### PR TITLE
Convert rhnactionvirtcreate data into one details column

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Change the way the virtual machine create / update action data are stored in the database. Instead of storing each parameter in its own column plus one-to-many relations for disks and interfaces, store it all as JSON in a `details` column.

This makes the action easier to extend without touching the database in the future and cleans up the schema a little bit.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: already covered

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/15393

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: schema/spacewalk/common/tables/rhnActionVirtCreate.sql [hunk 1]
```diff
@@ -21,20 +21,7 @@ CREATE TABLE rhnActionVirtCreate
...
-    vcpus                NUMERIC,
-    arch                 VARCHAR(20),
-    graphics_type        VARCHAR(20),
-    remove_disks         CHAR(1),
-    remove_interfaces    CHAR(1),
-    cobbler_system       VARCHAR(256),
-    kickstart_host       VARCHAR(256),
-    kernel_options       VARCHAR(2048),
-    cluster_definitions  VARCHAR(2048)
+    details              VARCHAR(2048)
```
Comment (mcalmer) [inline schema/spacewalk/common/tables/rhnActionVirtCreate.sql:24]: What about using "TEXT" to have no limit ?

File: schema/spacewalk/upgrade/susemanager-schema-4.3.0-to-susemanager-schema-4.3.1/200-rhnactionvirtcreate-refactoring.sql [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                DROP COLUMN IF EXISTS vcpus,
+                DROP COLUMN IF EXISTS arch CASCADE,
+                DROP COLUMN IF EXISTS graphics_type CASCADE,
+                DROP COLUMN IF EXISTS remove_disks,
+                DROP COLUMN IF EXISTS remove_interfaces,
+                DROP COLUMN IF EXISTS cobbler_system CASCADE,
+                DROP COLUMN IF EXISTS kickstart_host CASCADE,
+                DROP COLUMN IF EXISTS kernel_options CASCADE,
+                DROP COLUMN IF EXISTS cluster_definitions;
+            DROP TABLE IF EXISTS rhnactionvirtcreatediskdetails, rhnactionvirtcreateinterfacedetails;
```
Comment (mcalmer) [inline schema/spacewalk/upgrade/susemanager-schema-4.3.0-to-susemanager-schema-4.3.1/200-rhnactionvirtcreate-refactoring.sql:83]: seems these tables have sequences and indexes. Check if they need to be dropped as well.

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: schema/spacewalk/common/tables/rhnActionVirtCreate.sql (hunk 1 above)
Comment (cbosdo) [reply schema/spacewalk/common/tables/rhnActionVirtCreate.sql:24]: Oh! I didn't even know that SQL type. I'll use it then

File: schema/spacewalk/upgrade/susemanager-schema-4.3.0-to-susemanager-schema-4.3.1/200-rhnactionvirtcreate-refactoring.sql (hunk 2 above)
Comment (cbosdo) [reply schema/spacewalk/upgrade/susemanager-schema-4.3.0-to-susemanager-schema-4.3.1/200-rhnactionvirtcreate-refactoring.sql:83]: Right. I usually forget about them.

**Pull request merged**